rohmu/notifier/interface.py

```diff
@@ -0,0 +1,33 @@
+"""Copyright (c) 2022 Aiven, Helsinki, Finland. https://aiven.io/"""
+from abc import ABC, abstractmethod
+from typing import Optional
+
+
+class Notifier(ABC):
```

Comment:
Given the resource leaks in `(Backgrond)HTTPNotifier`, this interface should probably include at least a `stop` function to cleanup resources.

Reply:
Candiate for becoming a context manager too at this point?

Reply:
Although I'm thinking it's nothing inherent in the concept of a `Notifier` that really make a "resource" thingy... maybe just the `(Background)HTTPNotifier` class should implement that (and be accordingly documented) ?

Reply:
Would be annoying if the caller has to sniff the exact `Notifier` subclass(es) generated by `get_transfer` to decide if it should call `close`. It's better to have `close` part of the interface even if it sometimes does nothing, and it will give a subtle hint to the future implementer of another `Notifier` class that it's a good idea to think about cleanup :slightly_smiling_face:

Reply:
added `close` to the interface.